### PR TITLE
Added a new chem, Randonium

### PR DESCRIPTION
**Pull request by Jujumatic**
this is a new chem that can be made by mixing lean, meth and teslium and putting it in the pressure thing from lcass

this starts giving you every chem under the sun and gives you 5u of a random chem every single tick. the chances of you surviving to dying are about 40:60 and it's super fun to fuck with actually, if you don't mind risking dying

on mix you get a minor explosion, sparks and a tesla bolt that shoots around. deals a bit of damage but shouldn't be a super big inconvenience unless you're already near crit

i expect this to be a very fun chemical for admins to use

in order to make it truly random, no chemicals are blacklisted. admins be careful this means romerol can potentially pop up (though in about 10 tests i did, i ended up dying before it popped up)

## Changelog
:cl:
add: Added a new chemical, Randonium. This chemical has the property of fabricating other random chemicals while inside of you. Admins beware!
/:cl:

### CODE REVIEW

**Comment by Carbonhell** [comment]
I'd be ok with it being an admin only chem, but players getting it means they literally can spawn any sort of chem, even the hardest ones to make, by exploiting this one. I don't like the idea at all.

**Comment by Jujumatic** [comment]
i mean yeah they can get some amazing chems but good luck getting it in time before you die

**Comment by mikolaj007** [comment]
Mostly feels like some cruel and unusual execution method than something that could have a feasible use.

**Comment by Carbonhell** [comment]
> i mean yeah they can get some amazing chems but good luck getting it in time before you die

You aren't forced to use the chem on yourself though.

**Comment by mikolaj007** [comment]
Ye, pump someone, watch them die then starting removing chems from their body
The only problem is how diluted everything would be
only use i can forsee rn is getting mut toxins or some other chems which need few units to be usefull

**Comment by Jujumatic** [comment]
smoke would make big chempiles due to the amount of chemicals we're talking about here, however it'd be so diluted you might be able to get 0.1u of a specific chem if you were to scoop up from a few chempiles. unless you get something that needs a very small amount to work, it's useless. and you'd only be able to use it once, you can't dupe something less than around 10u

**Comment by DaydreamPuppeh** [comment]
+1 seems really fun.

**Comment by DeltaFire15** [comment]
We need this... a new fun chem to admininject into catbeasts.

File: hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -89,3 +89,28 @@
...
+	if(!everycheminthefuckinguniverse.len)
+		for(var/reagent in subtypesof(/datum/reagent/))
+			var/datum/reagent/newreagent = reagent
+			everycheminthefuckinguniverse += initial(newreagent.id)
+	return
+
+/datum/reagent/randonium/on_mob_life(mob/living/M)
+	..()
+	var/selectedchem
+	selectedchem = pick(everycheminthefuckinguniverse)
```
Comment (yoyobatty) [inline hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm]: Just put var/ behind for 1 line condenser

**Comment by Lxtoofonea** [comment]
+1, if its .1u of each random chem seems fine tbh

**Comment by Jujumatic** [comment]
it's 5u every tick cause otherwise there's no fun to be had... even 2u it just gets out of the body too quickly to do anything fun

**Comment by Eagle-Eyess** [comment]
What’s the difference between this and toxic goop? Besides the fact the toxic goop doesn’t allow all chems it’s basically the same thing.

**Comment by Jujumatic** [comment]
This one is more fun cause it can be literally any chem

Also it tends to make you fat from giving you tons of drinks and milk and stuff

**Comment by TothBrush** [comment]
> smoke would make big chempiles due to the amount of chemicals we're talking about here, however it'd be so diluted you might be able to get 0.1u of a specific chem if you were to scoop up from a few chempiles. unless you get something that needs a very small amount to work, it's useless. and you'd only be able to use it once, you can't dupe something less than around 10u

Correct me if I'm wrong but couldn't someone with an odyssey scan the 0.1u of any chem and then shoot a monkey over and over with the syringe gun to get more?

**Comment by TothBrush** [comment]
> What’s the difference between this and toxic goop? Besides the fact the toxic goop doesn’t allow all chems it’s basically the same thing.

I checked a while back and toxic goop was broken and didn't do anything at all.

**Comment by mikolaj007** [comment]
+1 i guess

**Comment by Majick1** [comment]
If it's admin only then there's no point though, you can just spawn the chem you want offhand. I'll +1 because more options are good, though maybe this would be better as a medical only TC item.

File: hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 1 above)
Comment (ghost) [reply hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm]: fixed

**Comment by ghost** [comment]
Headmin and designer approved so I'm going to merge this stale thing. If any balance issues come up later it's always possible to solve them as they go instead of incessantly speculating.

**Review by ghost — DISMISSED** [review]

**Comment by ghost** [comment]
I assume @Jujumatic had a good reason for copying every single reagent into a list

File: hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 2]
```diff
@@ -89,3 +89,27 @@
...
+	can_synth = TRUE
+	var/static/list/everycheminthefuckinguniverse = list()
+
+/datum/reagent/randonium/on_mob_add(mob/living/M)
+	..()
+	if(!everycheminthefuckinguniverse.len)
+		for(var/reagent in subtypesof(/datum/reagent/))
+			var/datum/reagent/newreagent = reagent
+			everycheminthefuckinguniverse += initial(newreagent.id)
+	return
```
Comment (yoyobatty) [inline hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm]: return here why?

File: hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 3]
```diff
@@ -89,3 +89,27 @@
...
+	color = "#245e28"	//rgb(36, 94, 40)
+	metabolization_rate = 0.1
+	taste_description = "a horrible mistake"
+	can_synth = TRUE
+	var/static/list/everycheminthefuckinguniverse = list()
+
+/datum/reagent/randonium/on_mob_add(mob/living/M)
+	..()
+	if(!everycheminthefuckinguniverse.len)
+		for(var/reagent in subtypesof(/datum/reagent/))
```
Comment (yoyobatty) [inline hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm]: should be `/datum/reagent`

File: hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 4]
```diff
@@ -89,3 +89,27 @@
...
+			var/datum/reagent/newreagent = reagent
+			everycheminthefuckinguniverse += initial(newreagent.id)
+	return
+
+/datum/reagent/randonium/on_mob_life(mob/living/M)
+	..()
+	var/selectedchem = pick(everycheminthefuckinguniverse)
+	M.reagents.add_reagent(selectedchem, 5)
+	return
```
Comment (yoyobatty) [inline hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm]: return here why?

File: hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 5]
```diff
@@ -89,3 +89,27 @@
...
+	taste_description = "a horrible mistake"
+	can_synth = TRUE
+	var/static/list/everycheminthefuckinguniverse = list()
+
+/datum/reagent/randonium/on_mob_add(mob/living/M)
+	..()
+	if(!everycheminthefuckinguniverse.len)
+		for(var/reagent in subtypesof(/datum/reagent/))
+			var/datum/reagent/newreagent = reagent
+			everycheminthefuckinguniverse += initial(newreagent.id)
```
Comment (yoyobatty) [inline hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm]: why initial? the reagent id was never going to change

**Review by yoyobatty — DISMISSED** [review]

**Comment by AnthonyJNgo** [comment]
+1, will label appropriately. Please make the changes batty requested and we will be good to go.

File: hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 2 above)
Comment (ghost) [reply hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm]: Fixed

File: hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 3 above)
Comment (ghost) [reply hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm]: yep

File: hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 4 above)
Comment (ghost) [reply hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm]: fixed

File: hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 5 above)
Comment (ghost) [reply hippiestation/code/modules/reagents/chemistry/reagents/other_reagents.dm]: no more initial

**Comment by ghost** [comment]
@yoyobatty I removed everything you wanted removed
@Jujumatic can you test this to see if it works?

**Comment by AnthonyJNgo** [comment]
Why isnt this merged

**Comment by ghost** [comment]
Juju tested this (he didn't)

**Review by ghost — APPROVED** [review]

**Pull request merged**